alacritty/src/ipc.rs

```diff
@@ -1,25 +1,30 @@
 //! Alacritty socket IPC.
 
 use std::ffi::OsStr;
-use std::io::{Error as IoError, ErrorKind, Result as IoResult};
-use std::os::unix::net::UnixDatagram;
+use std::io::{BufRead, BufReader, Error as IoError, ErrorKind, Result as IoResult, Write};
+use std::os::unix::net::{UnixListener, UnixStream};
 use std::path::PathBuf;
 use std::{env, fs, process};
 
 use glutin::event_loop::EventLoopProxy;
 use log::warn;
+use serde::{Deserialize, Serialize};
 
 use alacritty_terminal::thread;
 
-use crate::cli::Options;
+use crate::cli::{Options, TerminalOptions};
 use crate::event::{Event, EventType};
 
-/// IPC socket message for creating a new window.
-pub const SOCKET_MESSAGE_CREATE_WINDOW: [u8; 1] = [1];
-
 /// Environment variable name for the IPC socket path.
 const ALACRITTY_SOCKET_ENV: &str = "ALACRITTY_SOCKET";
 
+const JSON_SEPARATOR: u8 = b'\n';
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub enum IPCMessage {
+    CreateWindow(TerminalOptions),
+}
```

Comment:
Some docs on this enum would be nice just to know what it's used for.

Also in Rust the correct capitalization is `IpcMessage`.